src/burnchains/burnchain.rs

```diff
@@ -464,12 +468,12 @@ impl Burnchain {
             .expect("Overflowed u64 in calculating expected sunset_burn")
     }
 
-    pub fn is_reward_cycle_start(&self, block_height: u64) -> bool {
-        if block_height <= (self.first_block_height + 1) {
-            // not a reward cycle start if we're the first block after genesis.
+    pub fn is_reward_cycle_start(&self, burn_height: u64, num_sortitions: u64) -> bool {
+        if num_sortitions == 0 {
```

Comment:
Good catch!

Reply:
I'm a little confused why `num_sortitions` is needed here?  First, what's wrong with declaring that the first reward cycle in the system is always a PoB reward cycle?  It doesn't seem to require any special-case handling.  Second, if we cannot do this for some reason, then why not have `if block_height <= (self.first_block_height + self.pox_consts.reward_cycle_length + 1)` as the check, and remove `num_sortitions` entirely?

Reply:
From my understanding, and what I saw in the codebase, `self.first_block_height` is a constant for specifying that the genesis block can be mined anytime **after** `first_block_height`. I was under the impression that ensuring that we already did process some sortitions would be a robust way to check the required restriction.

Reply:
The number of sortitions that have occurred is always equal to `current_burn_block_height - self.first_block_height`, because the sortition DB calculates a sortition for each burnchain block even if no Stacks block was mined.  Unless something has changed in SIP-007, reward cycles are defined as fixed intervals of burnchain blocks, starting from `self.first_block_height` (cc @kantai).

Reply:
Hmmm.

So the intention of this check was to ensure that we're not starting a "new" reward cycle 1 block after genesis, so that we didn't have a 1 block reward cycle (comprising just the genesis block), followed by a burn reward cycle. Using `num_sortitions` achieves somewhat close to the same thing, except in the event that the "first sortition winner" is chosen > 1 reward cycle after `first_block_height`. Thinking about this more, though, I think it should be fine to allow that case -- after all, the code needs to deal with the case that a reward cycle didn't have any miners anyways, and allowing this case makes it so that we don't need to pass `num_sortitions` into this method.

Reply:
@kantai ha, just saw your comment.
Should we also get rid of the check:
```

        if effective_height == 0 {
            debug!(
                "effective_height = {}, reward cycle length == {}",
                effective_height, pox_consts.reward_cycle_length
            );
            return Err(CoordinatorError::NotPrepareEndBlock);
        }
```
?

Reply:
Addressed with latest

Reply:
Ooops -- I think maybe I wasn't very clear. I think that the original checks should be preserved, so `is_reward_cycle_start` would be unchanged versus next, i.e.:

```
    pub fn is_reward_cycle_start(&self, burn_height: u64) -> bool {
        if burn <= (self.first_block_height + 1) {
            // not a reward cycle start if we're the first block after genesis.
            false
        } else {
            let effective_height = burn_height - self.first_block_height;
            // first block of the new reward cycle
            (effective_height % (self.pox_constants.reward_cycle_length as u64)) == 1
        }
    }
```

And this check should still exist:

```rust
        if effective_height == 0 {
            debug!(
                "effective_height = {}, reward cycle length == {}",
                effective_height, pox_consts.reward_cycle_length
            );
            return Err(CoordinatorError::NotPrepareEndBlock);
        }
```

Because that makes sure that we don't start a prepare "end" in the first block either (as that would cause the node to try and count prepare blocks before the `first_burn_block`)

Reply:
Thinking through this more -- it isn't essential that `is_reward_cycle_start` have this check -- it just means that there'll be an extra bit in the PoX bit vector, which isn't a problem in itself.

The other check though, should exist.

Reply:
Check introduced in https://github.com/blockstack/stacks-blockchain/pull/2233/commits/438e0e7ab9dca697ff1ae6d26f61f48b02a5bf00